vmdb/app/models/vim_performance_analysis.rb

```diff
@@ -62,26 +62,40 @@ def get_targets
       elsif @options[:target_tags]
         topts = @options[:target_tags]
         includes = topts[:compute_type].to_sym == :Host ? {:hardware => {}, :vms => {:hardware => {}}} : nil
+        search_options = {
+          :class            => topts[:compute_type].to_s,
+          :results_format   => :objects,
+          :include_for_find => includes,
+          :userid           => @options[:userid],
+          :miq_group_id     => @options[:miq_group_id],
+        }
         if topts[:compute_filter]
-          search = topts[:compute_filter].kind_of?(MiqSearch) ? topts[:compute_filter] : MiqSearch.find(topts[:compute_filter])
-          @compute, attrs = Rbac.search(:class => topts[:compute_type].to_s ,:filter => search.filter, :results_format => :objects, :include_for_find => includes, :userid => @options[:userid], :miq_group_id => @options[:miq_group_id])
+          search = if topts[:compute_filter].kind_of?(MiqSearch)
+                     topts[:compute_filter]
+                   else
+                     MiqSearch.find(topts[:compute_filter])
+                   end
+          search_options[:filter] = search.filter
         elsif topts[:compute_tags]
-          @compute, attrs = Rbac.search(:class => topts[:compute_type].to_s ,:tag_filters => {"managed" => topts[:compute_tags]}, :results_format => :objects, :include_for_find => includes, :userid => @options[:userid], :miq_group_id => @options[:miq_group_id])
-        else
-          @compute, attrs = Rbac.search(:class => topts[:compute_type].to_s ,:results_format => :objects, :include_for_find => includes, :userid => @options[:userid], :miq_group_id => @options[:miq_group_id])
+          search_options[:tag_filters] = {"managed" => topts[:compute_tags]}
         end
+        @compute, _attrs = Rbac.search(search_options)
 
         MiqPreloader.preload(@compute, :storages)
-        stores = @compute.collect {|c| storages_for_compute_target(c)}.flatten.uniq
+        stores = @compute.collect { |c| storages_for_compute_target(c) }.flatten.uniq
 
+        filter_options = {:class => Storage, :userid => @options[:userid], :miq_group_id => @options[:miq_group_id]}
         if topts[:storage_filter]
-          search = topts[:storage_filter].kind_of?(MiqSearch) ? topts[:storage_filter] : MiqSearch.find(topts[:compute_filter])
```

Comment:
The rest of this commit should be a super-conservative A :: B translation, with very little room for behavioural variations to creep in.

However, I've actually changed a thing here: previously, this line referenced `:compute_filter`, which I'm 99% sure is a bug. So I've changed it to the more likely-sounding `:storage_filter` below.